### DISCUSSION TITLE
QR for your WiFi access

### DISCUSSION THREAD

**Discussion by barbuuu** [Blackymas/NSPanel_HA_Blueprint#305] [Show and tell]
If you want to display QR code representing your WiFi credentials for visitors, then you have to add this line into blueprint "QR Code content" field:
WIFI:S:{yourssid};T:{encriptiontype};P:{yourwifipass};;

example: for SSID: DAN with Encription Type: WPA2 with Wifi password: secret123
WIFI:S:DAN;T:WPA2;P:secret123;;
![QRforWiFi](https://user-images.githubusercontent.com/44302559/216945159-d0c3d2fe-cedc-467d-9615-1a7841111b6d.png)

Enjoy! and don't forgot to Activate the QR Code on the Blueprint :)

**Comment by Blackymas** [comment]
move it to show and tell

Thanks

**Comment by Statusbit** [comment]
Perfekt - Thanks a lot!

**Comment by MarkuZP** [comment]
Is it possible that some characters are not allowed or lenght of the SSID is limited?
My guest WLAN has a longer name with an ! in the middle and its not working
SSID: FRITZ! the cat Gastzugang

**Reply by deejaybeam** [reply]
From the release notes: "For the QR Code Content you have 50 characters at your disposal which you can use freely."
So check the length of the resulting string

**Reply by MarkuZP** [reply]
"FRITZ! the cat Gastzugang" is under 50 :-) But still not working...

**Reply by deejaybeam** [reply]
But you need the full string like:
"WIFI:S:FRITZ! the cat Gastzugang;T:WPA;P:DeinGeheimesSuperPasswortFürDenGastzugang;;"
If you only send the SSID this could not work

**Reply by MarkuZP** [reply]
> But you need the full string like: "WIFI:S:FRITZ! the cat Gastzugang;T:WPA;P:DeinGeheimesSuperPasswortFürDenGastzugang;;" If you only send the SSID this could not work

Yes that was clear! :-). The problem is the "! " I guess. i tried only "Gastzugang" and it worked.

**Comment by Statusbit** [comment]
blank characters? or %20?

**Comment by barbuuu** [comment]
Not sure if someone is interested but the qrCode value is evaluated as a homeassistant Template, so it is possible to get values from HomeAssistant within the qrCode.

"WIFI:S:{{states('input_text.test_ssid')}};T:WPA;P:{{states('input_text.test_pw')}};;"

**Reply by stevoh6** [reply]
Yeah, bit it think/hope that lot of user using separate _iot wifi for Ha and related stuff :)

**Comment by MichalAug** [comment]
Is it possible to trigger the display of the qr code?
I would like to display it at the entrance, when the door is opened.

**Comment by hollabee** [comment]
Yes, that's possible, check here:
https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know#11-call-certain-pages-directly

You would need an automation triggering the service, when your door is opened. Done :)

**Reply by MichalAug** [reply]
f**king brilliant, thanks!

**Reply by MichalAug** [reply]
> Yes, that's possible, check here: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know#11-call-certain-pages-directly
> 
> You would need an automation triggering the service, when your door is opened. Done :)

My euphoria did not last long. I guess I'm doing something wrong.
Calling qrcode causes a blank white square to appear on the screen.
Of course triggering the qrcode from the screen works fine, and remotely it looks like there is no data for that qrcode. Do you have to add them somehow when invoking the "printf" command?

**Comment by hollabee** [comment]
Just tried it myself:
- I have a QR Code defined in the Blueprint
- Opening it on the screen directly by touching it: works
- Sending the following command:

`service: esphome.nspanel_send_command_printf
data:
  cmd: page qrcode`

Opens the qrcode page with my QR Code. So this works for me.

The code simply "forces" the nspanel to open the specific page, the information on each page is not being transmitted using the service call. 

So my guess is, that there is something odd with your QR Code. 

Did you update all things, Panel, ESPHome and the Blueprint correctly?

**Comment by MichalAug** [comment]
I checked everything again. I updated everything, changed language, changed qr code data.
From the icon it works, from the automation it does not work. Neither the qr code nor the description of this code. There is only a white square.
Other tabs like entitypage01 work fine.

I send data with node-red: {"cmd":"page qrcode"} it causes a white screen.
{"cmd":"page entitypage01"} displays the entities from screen 01 normally.

**Reply by MichalAug** [reply]
Update:
There is a other service: nspanel_qr_code, which works, in data you need to give qr code data, it has nothing to do with blueprint code. 
I don't know how to add a description to it at the top, but other than that it works fine.

**Reply by edwardtfn** [reply]
> I don't know how to add a description to it at the top, but other than that it works fine.

Try this:
```yaml 
service: esphome.nspanel_send_command_printf
data:
  component: "qrcode_label"
  message: "Your description"
```
